src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -894,30 +900,53 @@ internal virtual void PrepareSession()
                 // Store the UserAgent string
                 WebSession.UserAgent = UserAgent;
             }
-
-            if (Proxy is not null)
+            // Proxy and NoProxy parameters are mutually exclusive
+            // If NoProxy is provided, WebSession will turn off the proxy
+            // and if Proxy is provided NoProxy will be turned off
+            if (NoProxy.IsPresent)
             {
-                WebProxy webProxy = new(Proxy);
-                webProxy.BypassProxyOnLocal = false;
-                if (ProxyCredential is not null)
-                {
-                    webProxy.Credentials = ProxyCredential.GetNetworkCredential();
-                }
-                else if (ProxyUseDefaultCredentials)
+                WebSession.NoProxy = true;
+            }
+            else
+            {
+                if (Proxy is not null)
                 {
-                    // If both ProxyCredential and ProxyUseDefaultCredentials are passed,
-                    // UseDefaultCredentials will overwrite the supplied credentials.
-                    webProxy.UseDefaultCredentials = true;
+                    WebProxy webProxy = new(Proxy);
+                    webProxy.BypassProxyOnLocal = false;
```

Comment:
/cc @CarloToso It seems we don't use `BypassProxyOnLocal`. If you have an interest please open new tracking issue for follow investigations.

Reply:
`BypassProxyOnLocal==true` can be set if we add a new Proxy to the WebSession and then we use it.
Do you think we should set `BypassProxyOnLocal true or false` directly from WebRequestPSCmdlet (if so I'll open a tracking issue)?

Reply:
@CarloToso I say we never use a value of the BypassProxyOnLocal property.